readthedocs/doc_builder/environments.py

```diff
@@ -267,23 +268,39 @@ class BuildEnvironment(object):
 
     Base class for wrapping command execution for build steps. This provides a
     context for command execution and reporting, and eventually performs updates
-    on the build object itself, reporting success/failure, as well as top-level
-    failures.
+    on the build object itself, reporting success/failure, as well as failures
+    during the context manager enter and exit.
+
+    Any exceptions raised inside this context and handled by the eventual
+    :py:meth:`__exit__` method, specifically, inside :py:meth:`handle_exception`
+    and :py:meth:`update_build`. If the exception is a subclass of
+    :py:cls:`BuildEnvironmentError`, then this error message is added to the
+    build object and is shown to the user as the top-level failure reason for
+    why the build failed. Other exceptions raise a general failure warning on
+    the build.
+
+    We only update the build through the API in one of three cases:
+
+    * The build is not done and needs incremental builds
+    * The build failed and we should always report this change
+    * The build was successful and ``commit`` is ``True``
 
     :param project: Project that is being built
     :param version: Project version that is being built
     :param build: Build instance
     :param record: Record status of build object
     :param environment: shell environment variables
+    :param commit: update the build object via API if the build was successful
```

Comment:
I think this is still confusing.

When `commit=True` (the default) and we are in the middle of the build, the build won't be updated. So, I think this should have a better name (if that is possible) like `last_command`, since I think it's more representative of what it measn.

Anyway, I don't see the best solution very clear.

Reply:
probably the most explicit we can is `update_on_success`